acceptance/tests/verify_facts.rb

```diff
@@ -29,9 +29,7 @@ def validate_fact(name, node, fact_value, hidden)
               # YAML-CPP seems to drop the decimal whenever it feels like it, so just match Int too.
               fact_value.is_a? Float or fact_value.is_a? Integer or fact_value.to_s =~ /^(\.inf|\.nan|-\.inf)$/
             when 'string'
-              # Any string can be interpreted as a more specific type by ruby
-              # YAML.
-              true
+              fact_value.nil? or fact_value.is_a? String or fact_value.is_a? TrueValue or fact_value.is_a? FalseValue
```

Comment:
nit, it looks like `validate_fact` will never be called with a nil `fact_value`? https://github.com/MikaelSmith/facter/blob/FACT-1317/acceptance/tests/verify_facts.rb#L68